packages/yarn-plugin-allow-scripts/test/index.spec.mts

```diff
@@ -0,0 +1,238 @@
+ */
+const run = (
+  t: any,
+  args: string[],
+  cwd: string,
+  env: {} = {},
+): {exitCode: number|null, out: string, err: string, output: (string|Buffer|null)[], signal: string|null, error?: Error|undefined} => {
+  try {
+    const defaultOpts = realisticEnvOptions(cwd);
+    const result = spawnSync(
+      isWindows
+        ? process.env.COMSPEC || 'cmd.exe'
+        : process.env.SHELL || '/bin/sh',
+      [
+        ...(isWindows ? ['/d', '/s', '/c'] : ['-c']),
+        [
+          isWindows ? 'corepack.cmd' : 'corepack',
+          'yarn@4',
+          ...args
+        ].join(' '),
+      ],
+      {
+        ...defaultOpts,
+        env: {
+          ...process.env,
+          ...defaultOpts.env,
+          ...env,
+        }
+      },
+    )
+    return {
+      exitCode: result.status,
+      out: result?.stdout?.toString()?.trim(),
+      err: result?.stderr?.toString()?.trim(),
+      output: result?.output,
+      signal: result.signal,
+      error: result.error,
+    }
+  } catch(error) {
+    console.error(error);
+    t.fail(
+      `Failed calling '${args.join(' ')}': ${JSON.stringify(
+        {
+          cwd,
+          error,
+        },
+        undefined,
+        2
+      )}`
+    )
+    return {
+      error,
+      output: [],
+      exitCode: null,
+      signal: null,
+      out: '',
+      err: '',
+    };
+  }
+}
+
+test('install - errors when adding dependency with unspecified preinstall script', async (t: any) => {
+  const projectName = 'uninitialized';
+  const projectRoot = join(dirname(import.meta.url.replace(/^file:/, '')), 'projects', projectName);
+
+  try {
+    // delete any leftover test artifacts
+    rmSync(join(projectRoot, PACKAGE_JSON), { force: true })
+
+    // init project
+    const initRes = run(t, ['init', '-y'], projectRoot)
+    t.is(initRes.exitCode, 0);
+
+    // install the plugin
+    const importRes = run(t, ['plugin', 'import', normalize('../../../bundles/@yarnpkg/plugin-allow-scripts.js')], projectRoot)
+    t.is(importRes.err, '');
+    t.regex(importRes.out, /YN0000: Saving the new plugin in .yarn\/plugins\/@yarnpkg\/plugin-allow-scripts.cjs/);
+    t.is(importRes.exitCode, 0);
+
+    const addRes = run(t, ['add', '@lavamoat/preinstall-always-fail'], projectRoot)
+    t.regex(addRes.err, /allow-scripts detected attempted execution of unconfigured package script. \["@lavamoat\/preinstall-always-fail","preinstall"/);
+    t.is(addRes.exitCode, 1);
+
+    // trigger the plugin
+    const installRes = run(t, ['install'], projectRoot)
+    t.regex(installRes.out, /packages missing configuration:\n- @lavamoat\/preinstall-always-fail/);
+    t.regex(installRes.err, /@lavamoat\/allow-scripts has detected dependencies without configuration. explicit configuration required./);
+    t.is(installRes.exitCode, 1);
+  } finally {
+    cleanup(projectName);
+  }
+})
+
+test('install - allows adding and installing package with specified preinstall script', async (t: any) => {
```

Comment:
Looking at the 'allowed' folder in fixtures, I don't see what this is actually allowing. 

https://github.com/LavaMoat/LavaMoat/pull/1218/files#diff-531f1c2b60c68ac2339c35184659e364dab32ce1431bcae05c136e6d4cfa4a8eR6 says the preinstall is not allowed. 

What's being tested here?

Reply:
The end-result is we're allowed to run `install` because `preinstall-always-fail` is handled correctly. Maybe there is a better name?

Reply:
Updated this test-case and its description